test/one/basic_test.go

```diff
@@ -0,0 +1,182 @@
+package one
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	aws "github.com/gruntwork-io/terratest/modules/aws"
+	g "github.com/gruntwork-io/terratest/modules/git"
+	"github.com/gruntwork-io/terratest/modules/ssh"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	util "github.com/rancher/terraform-provider-rancher2/test"
+)
+
+func TestOneBasic(t *testing.T) {
+	t.Parallel()
+	id := util.GetId()
+	region := util.GetRegion()
+	directory := "one"
+	owner := "terraform-ci@suse.com"
+	util.SetAcmeServer()
+
+	repoRoot, err := filepath.Abs(g.GetRepoRoot(t))
+	if err != nil {
+		t.Fatalf("Error getting git root directory: %v", err)
+	}
+
+	exampleDir := repoRoot + "/examples/use-cases/" + directory
+	testDir := repoRoot + "/test/data/" + id
+
+	err = util.CreateTestDirectories(t, id)
+	if err != nil {
+		os.RemoveAll(testDir)
+		t.Fatalf("Error creating test data directories: %s", err)
+	}
+	keyPair, err := util.CreateKeypair(t, region, owner, id)
+	if err != nil {
+		os.RemoveAll(testDir)
+		t.Fatalf("Error creating test key pair: %s", err)
+	}
+	err = os.WriteFile(testDir+"/id_rsa", []byte(keyPair.KeyPair.PrivateKey), 0600)
+	if err != nil {
+		err = aws.DeleteEC2KeyPairE(t, keyPair)
+		if err != nil {
+			t.Logf("Failed to destroy key pair: %v", err)
+		}
+		os.RemoveAll(testDir)
+		t.Fatalf("Error creating test key pair: %s", err)
+	}
+	sshAgent := ssh.SshAgentWithKeyPair(t, keyPair.KeyPair)
+	t.Logf("Key %s created and added to agent", keyPair.Name)
+
```

Comment:
I'll propose creating a `TestConfig` struct and a `NewTestConfig` helper function to encapsulate this common setup.